components/range_cache_memory_engine/src/range_manager.rs

```diff
@@ -71,124 +129,185 @@ impl RangeMeta {
         self.safe_point = safe_point;
     }
 
-    fn derive_from(id: u64, r: &RangeMeta) -> Self {
-        Self {
-            id,
-            range_snapshot_list: SnapshotList::default(),
-            safe_point: r.safe_point,
-        }
+    pub(crate) fn get_range(&self) -> &CacheRange {
+        &self.range
     }
 
-    pub(crate) fn range_snapshot_list(&self) -> &SnapshotList {
-        &self.range_snapshot_list
+    pub fn get_state(&self) -> RegionState {
+        self.state
     }
-}
 
-#[derive(Default)]
-struct IdAllocator(u64);
+    pub(crate) fn set_state(&mut self, new_state: RegionState) {
+        self.state = new_state;
```

Comment:
Do we need to verify the new_state must obey some rules?

Reply:
Done